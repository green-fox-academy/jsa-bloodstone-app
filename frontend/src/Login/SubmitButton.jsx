import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, TouchableHighlight } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import Colors from '../common/colors';

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 50,
    width: 60,
    height: 60,
    backgroundColor: Colors.tealColor,
  },
  icon: {
    marginRight: -2,
    marginTop: -2,
  },
});

function SubmitButton({ onPress }) {
  return (
    <TouchableHighlight
      style={styles.button}
      underlayColor={Colors.lightenTeal}
      onPress={onPress}
    >
      <Icon
        name="angle-right"
        color="#fff"
        size={32}
        style={styles.icon}
      />
    </TouchableHighlight>
  );
}

SubmitButton.propTypes = {
  onPress: PropTypes.func,
};

SubmitButton.defaultProps = {
  onPress: null,
};

export default SubmitButton;
