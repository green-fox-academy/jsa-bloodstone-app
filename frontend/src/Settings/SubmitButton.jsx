import React from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet, TouchableHighlight, Text, ViewPropTypes,
} from 'react-native';
import Colors from '../common/colors';

function SubmitButton({
  style, onPress, disabled, text,
}) {
  const styles = StyleSheet.create({
    button: {
      borderRadius: 5,
      width: 100,
      padding: 10,
      backgroundColor: disabled ? Colors.greyColor : Colors.tealColor,
    },
    text: {
      fontSize: 16,
      color: Colors.whiteColor,
      textAlign: 'center',
    },
  });

  return (
    <TouchableHighlight
      style={[styles.button, style]}
      underlayColor={Colors.lightenTeal}
      onPress={onPress}
      disabled={disabled}
    >
      <Text style={styles.text}>{text}</Text>
    </TouchableHighlight>
  );
}

SubmitButton.propTypes = {
  style: ViewPropTypes.style,
  onPress: PropTypes.func,
  disabled: PropTypes.bool,
  text: PropTypes.string.isRequired,
};

SubmitButton.defaultProps = {
  onPress: null,
  disabled: false,
  style: null,
};

export default SubmitButton;
