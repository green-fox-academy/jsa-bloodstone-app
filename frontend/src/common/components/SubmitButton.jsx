import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, TouchableHighlight } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import Colors from '../colors';

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 50,
    width: 50,
    height: 50,
    backgroundColor: Colors.blueColor,
  },
});

function SubmitButton({ onPress, direction }) {
  return (
    <TouchableHighlight
      style={styles.button}
      underlayColor={Colors.lightenTeal}
      onPress={onPress}
    >
      <Ionicons
        name={`ios-arrow-round-${direction}`}
        color="#fff"
        size={36}
      />
    </TouchableHighlight>
  );
}

SubmitButton.propTypes = {
  onPress: PropTypes.func,
  direction: PropTypes.oneOf(['back', 'forward']),
};

SubmitButton.defaultProps = {
  onPress: null,
  direction: 'forward',
};

export default SubmitButton;
