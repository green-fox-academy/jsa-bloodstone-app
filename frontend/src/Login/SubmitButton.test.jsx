import React from 'react';
import renderer from 'react-test-renderer';

import SubmitButton from './SubmitButton';

jest.mock('react-redux');
jest.mock('./actionCreator');

describe('<SubmitButton />', () => {
  it('Should render buildings and match the snapshot', () => {
    const component = renderer.create(<SubmitButton />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
