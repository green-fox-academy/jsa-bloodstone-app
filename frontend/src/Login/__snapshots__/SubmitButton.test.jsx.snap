// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> Should render buildings and match the snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#10978b",
      "borderRadius": 50,
      "height": 60,
      "justifyContent": "center",
      "width": 60,
    }
  }
>
  <Text />
</View>
`;
